R/datalog.R

```diff
@@ -44,6 +47,13 @@ parse_expression <- function(s) {
             ret <- list(op, v[[1]], v[[2]])
         }
 
+        #Vector: processing vectors written as c(?a, ?b) to list(?a, ?b)
+        if(startsWith(ret, "c(")){
+            m <- gsub(" ", "", ret)
+            m <- stringr::str_match(m, "c(.*)\\((.*)\\)")
```

Comment:
Does this regex work only for vector of two elements, or for arbitrary-length ones as well?

Reply:
Works for arbitrary length. This particular regex returns whats in between `c(` and `)`.

```
qt <- query(find(?a, ?b),
            where(
                d(untuple(?x), c(?a, ?b, ?c, ?d)
                  ))) 
```
resolves to:
`{"query":{"find":["?a","?b"],"where":[[["untuple","?x"],["?a","?b","?c","?d"]]]}} 
`